src/main/java/org/springframework/retry/backoff/ExponentialBackOffPolicy.java

```diff
@@ -289,15 +289,27 @@ public synchronized long getSleepAndIncrement() {
 		}
 
 		protected long getNextInterval() {
-			return (long) (this.interval * getMultiplier());
+			if (initialIntervalSupplier == null) {
+				return (long) (this.interval * getMultiplier());
+			}
+			else {
+				return this.interval == DEFAULT_INITIAL_INTERVAL
+						? (long) (this.initialIntervalSupplier.get() * getMultiplier())
+						: (long) (this.interval * getMultiplier());
+			}
 		}
 
 		public double getMultiplier() {
 			return this.multiplierSupplier != null ? this.multiplierSupplier.get() : this.multiplier;
 		}
 
 		public long getInterval() {
-			return this.intervalSupplier != null ? this.intervalSupplier.get() : this.interval;
+			if (initialIntervalSupplier == null) {
+				return this.interval;
+			}
+			else {
+				return this.interval == DEFAULT_INITIAL_INTERVAL ? this.initialIntervalSupplier.get() : this.interval;
```

Comment:
Don't we need to call `intervalSupplier` every time?
Looks like with your change we do that only once.

Reply:
Only first time when we need initial value, further iteration store multiplayer value in the `interval` variable [here ](https://github.com/spring-projects/spring-retry/blob/903640ec21c02a6278e1d9415edb59f3f1f5c0be/src/main/java/org/springframework/retry/backoff/ExponentialBackOffPolicy.java#L286)